docs/cql/ddl.rst

```diff
@@ -116,7 +116,7 @@ name                 kind       mandatory   default   description
                                                       details below).
 ``durable_writes``   *simple*   no          true      Whether to use the commit log for updates on this keyspace
                                                       (disable this option at your own risk!).
-``tablets``          *map*      no                    Experimental - enables tablets for this keyspace (see :ref:`tablets<tablets>`)
+``tablets``          *map*      no                    Enables or disables tablets for the keyspace (see :ref:`tablets<tablets>`)
```

Comment:
default?

Reply:
The default depends on the the global `enable_tablets` config option (that is enabled by default for new clusters, and disabled by default when upgrading existing clusters).

New keyspaces will have `tablets` enabled by default if and only if the global `enable_tablets` option is set to `true` and the whole cluster runs a version supporting tablets (6.0 or later).

Reply:
> Is the above correct?

Yes.  Right on the spot.
Just requires rolling restart after enabling tablets in scylla.yaml (unless done along with rolling upgrade, that entails a restart of each node anyhow)

@xemul is it live-updateable?
Can the modified option be updated by the scylla process using SIGHUP, with no need for restart?

Reply:
> > Is the above correct?
> 
> Yes. Right on the spot. Just requires rolling restart after enabling tablets in scylla.yaml (unless done along with rolling upgrade, that entails a restart of each node anyhow)
> 

@bhalevy but still even with `enable_tablets=true` I can use tablets (if Scylla supports it) by using `tablets` option set to true in `CREATE KEYSPACE`?

Reply:
> > > Is the above correct?
> > 
> > 
> > Yes. Right on the spot. Just requires rolling restart after enabling tablets in scylla.yaml (unless done along with rolling upgrade, that entails a restart of each node anyhow)
> 
> @bhalevy but still even with `enable_tablets=false` I can use tablets (if Scylla supports it) by using `tablets` option set to true in `CREATE KEYSPACE`?

No.  With `enable_tablets=false` you shouldn't be able to create a keyspace with `tablets = {'enabled':true}` or  `tablets = {'initial':N}`.  If the query does not return an error, please open an issue about it.  It should be rejected.

Reply:
@xemul please confirm the above.

Reply:
I'm not sure what would happen to existing keyspace that may have already been created with tablets enabled after restarting with `tablets_enabled: false`.  It's possible that they wouldn't be accessible, or just that their tablet map will be static and never change.  I don't know how they will be treated in reaction to topology changes like adding or decommissioning nodes, since the load balancer might not be started at all when tablets are disabled.
It might take a long time for us to support migration from tablets back to vnodes, so once keyspaces are created with tablets enabled, going back to vnodes might require deleting those keyspace before disabling tablets.

Reply:
> @xemul is it live-updateable?
> Can the modified option be updated by the scylla process using SIGHUP, with no need for restart?

It's not live-updateable. Enabling tablets on the fly would require re-negotiating features, subscribe tablets allocator to migration events, add system.tablets table, and none of it can be done runtime nowadays

Reply:
> No. With enable_tablets=false you shouldn't be able to create a keyspace with tablets = {'enabled':true} or tablets = {'initial':N}. If the query does not return an error, please open an issue about it. It should be rejected.

Yes, that's almost correct, CREATE KEYSPACE should throw in case tablets are not _negotiated_ via feature service, but user asks for it

```c++
void tablet_aware_replication_strategy::validate_tablet_options(const abstract_replication_strategy& ars,
                                                                const gms::feature_service& fs,
                                                                const replication_strategy_config_options& opts) const {
    if (ars._uses_tablets && !fs.tablets) {
        throw exceptions::configuration_exception("Tablet replication is not enabled");
    }   
}    
```

In turn, if tablets are not enabled by scylla.yaml, scylla node will _not_ try to negotiate tablets feature

Reply:
> I'm not sure what would happen to existing keyspace that may have already been created with tablets enabled after restarting with tablets_enabled: false.

They should continue to operate in vnodes mode. When loading schema, scylla checks the system.scylla_keyspaces table's initial_tablets column. If it's empty, the keyspace is considered to be vnodes-based regardless of what

Reply:
so, to summarise, if `enable_tablets=false` then we cannot create keyspace using tablets options. In case `enable_tablets=true` it means tablets are default choice and in case user doesn't want to use it, need to use `tablets = {'enabled': false}` - i think we should state this in this form to be clear.

Reply:
> > I'm not sure what would happen to existing keyspace that may have already been created with tablets enabled after restarting with tablets_enabled: false.
> 
> They should continue to operate in vnodes mode. When loading schema, scylla checks the system.scylla_keyspaces table's initial_tablets column. If it's empty, the keyspace is considered to be vnodes-based regardless of what

If the existing tables have tablets enabled, they cannot be accessed safely in vnodes mode.
Writing will create a big mess, and reading is very likely to miss data that is not stored in the expected location based on the vnode-based replication map.

Reply:
>  In case enable_tablets=true it means tablets are default choice

Provided other nodes in the cluster also have it so the tablets feature was negotiated and enabled

Reply:
> > > I'm not sure what would happen to existing keyspace that may have already been created with tablets enabled after restarting with tablets_enabled: false.
> > 
> > 
> > They should continue to operate in vnodes mode. When loading schema, scylla checks the system.scylla_keyspaces table's initial_tablets column. If it's empty, the keyspace is considered to be vnodes-based regardless of what
> 
> If the existing tables have tablets enabled, they cannot be accessed safely in vnodes mode. Writing will create a big mess, and reading is very likely to miss data that is not stored in the expected location based on the vnode-based replication map.

Sorry, I mis-read your question :( The answer is symmetrical -- on boot scylla will pick its replication strategy an options from schema tables, will try to turn tablets ON and will fail